test/jdk/javax/swing/JMenuItem/bug4839464.java

```diff
@@ -0,0 +1,242 @@
+                        action.putValue(Action.NAME, "Second Name");
+                    } else {
+                        action.putValue(Action.NAME, "First Name");
+                    }
+                }
+            });
+
+            changeMnemonicButton.addActionListener(new AbstractAction() {
+                public void actionPerformed(ActionEvent e) {
+                    Integer mnem = (Integer) action.getValue(Action.MNEMONIC_KEY);
+                    if (mnem.intValue() == 0) {
+                        action.putValue(Action.MNEMONIC_KEY, new Integer('N'));
+                    } else {
+                        action.putValue(Action.MNEMONIC_KEY, new Integer(0));
+                    }
+                }
+            });
+
+            changeAcceleratorButton.addActionListener(new AbstractAction() {
+                public void actionPerformed(ActionEvent e) {
+                    if (action.getValue(Action.ACCELERATOR_KEY) == ks1) {
+                        action.putValue(Action.ACCELERATOR_KEY, ks2);
+                    } else {
+                        action.putValue(Action.ACCELERATOR_KEY, ks1);
+                    }
+                }
+            });
+
+            changeShortDescButton.addActionListener(new AbstractAction() {
+                public void actionPerformed(ActionEvent e) {
+                    String shortDescr = (String) action.getValue(Action.SHORT_DESCRIPTION);
+                    if ("Just a text".equals(shortDescr)) {
+                        action.putValue(Action.SHORT_DESCRIPTION, null);
+                    } else {
+                        action.putValue(Action.SHORT_DESCRIPTION, "Just a text");
+                    }
+                }
+            });
+
+            action.putValue(Action.NAME, "Second Name");
+            action.putValue(Action.MNEMONIC_KEY, new Integer('N'));
+            action.putValue(Action.ACCELERATOR_KEY, ks1);
+            action.putValue(Action.SHORT_DESCRIPTION, null);
+        });
+
+        r.delay(1000);
+        r.waitForIdle();
+        // Run tests
+        test();
+        r.delay(1000);
+        r.waitForIdle();
+
+        if (!passed) {
+            throw new RuntimeException(reason + "\nTest failed.");
+        }
+    }
+
+    public boolean compareObjects(Object a, Object b) {
+        if (a == null) {
+            return (b == null);
+        }
+        return a.equals(b);
+    }
+
+    // Actual tests
+    public void test() {
+        JRobot robo;
+        robo = JRobot.getRobot();
+        robo.delay(500);
+        Object tmpResult;
+
+        // Check Action.NAME handling
+        tmpResult = item.getText();
+        robo.clickMouseOn(changeNameButton);
+        if(compareObjects(tmpResult, item.getText())) {
+            passed = false;
+            reason = reason + "\n Action.NAME";
+        }
+
+        // Check mnemonics
+        int tmpInt = item.getMnemonic();
+        robo.clickMouseOn(changeMnemonicButton);
```

Comment:
@DamonGuy Util's JRobot is used for here

Reply:
@DamonGuy 'clickMouseOn()'. Earlier this test had a AWT Robot and JRobot. It has been changed recently to use AWT Robot only.

Reply:
Gotcha. Just for my understanding, does JRobot depend on Util? `clickMouseOn` is in JRobot and I thought Util and JRobot were separate since they're java files.